clouds/aws/roles/roles.py

```diff
@@ -19,6 +19,36 @@ def iam_client(region=DEFAULT_AWS_REGION):
     return boto3.client(service_name="iam", region_name=region)
 
 
+def get_roles(client=None):
+    """
+    Get all roles from IAM.
+    Args:
+        client: A boto3 client for IAM.
+
+    Returns:
+         List of roles.
+    """
+    LOGGER.info("Retrieving all roles from IAM.")
+
+    if not client:
+        client = iam_client()
+
+    marker = None
+    is_truncated = True
+    roles = []
+
+    while is_truncated:
```

Comment:
we may be here forever (maybe on some edge cases) but we can try to limit it.
call `roles = client.list_roles(MaxItems=1000) ` outside the `while` and only if you have `IsTruncated` then enter the `while`

```
response = client.list_rolesMaxItems=1000)
roles = response["Roles"]
is_truncated = response["IsTruncated"]
marker = response["Marker"]

if not is_truncated:
    return roles

while is_truncated:
    response = client.list_roles(Marker=marker, MaxItems=1000)
    roles.extend(response["Roles"])
    is_truncated = response["IsTruncated"]
    marker = response["Marker"]

return roles
```